app/models/embed/purl.rb

```diff
@@ -137,6 +138,30 @@ def initialize(msg = 'The requested PURL resource was not embeddable.')
         super
       end
     end
+
+    class Duration < ISO8601::Duration
+      def to_s
+        raise 'Duration#to_s does not support specifying durations in weeks' if atoms[:weeks] != 0
+        raise 'Duration#to_s does not support specifying negative durations' if atoms.values.detect { |val| val < 0 }
```

Comment:
Do we really want to raise an error in this case?  That would, AFAICT, cause the embed viewer to error out and not display anything for these objects.  Should we consider a more graceful approach for when the data is not what we want/expect?